<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <arg name="cmd_vel_topic" default="cmd_vel"/>
    <arg name="db_chassis_services_enabled" default="true"/>
    <arg name="db_chassis_publish_odom_tf" default="true"/>
    <arg name="db_chassis_use_sensor_msg_time" default="false"/>
    <arg name="db_chassis_odom_frame" default="odom"/>

    <group ns="dodobot">
        <node name="db_chassis" pkg="db_chassis" type="db_chassis_node" output="screen" required="true">
            <remap from="cmd_vel" to="$(arg cmd_vel_topic)" />

            <param name="ticks_per_rotation" value="1496.88"/>
            <param name="wheel_distance_mm" value="200.0"/>

            <param name="min_angular_speed" value="0.5"/>
            <!-- <param name="min_angular_speed" value="2.0"/> -->
            <param name="max_angular_speed" value="6.0"/>
            <param name="min_linear_speed" value="0.06"/>
            <param name="max_linear_speed" value="0.69"/>

            <param name="tilter_lower_angle_deg" value="-50.0"/>
            <param name="tilter_upper_angle_deg" value="0.0"/>
            <param name="tilter_lower_command" value="10"/>
            <param name="tilter_upper_command" value="140"/>

            <param name="services_enabled" value="$(arg db_chassis_services_enabled)"/>
            <!-- <param name="odom_estimate_method" value="runge-kutta"/> -->
            <!-- <param name="odom_estimate_method" value="geometric"/> -->
            <param name="publish_odom_tf" value="$(arg db_chassis_publish_odom_tf)"/>
            <param name="use_sensor_msg_time" value="$(arg db_chassis_use_sensor_msg_time)"/>
            <param name="odom_parent_frame" value="$(arg db_chassis_odom_frame)"/>
            <param name="idle_timeout" value="1.0"/>
        </node>
    </group>
</launch>
